src/library/scala/concurrent/duration/Duration.scala

```diff
@@ -124,6 +124,8 @@ object Duration {
       Undefined
     else if (nanos > Long.MaxValue || nanos < Long.MinValue)
       throw new IllegalArgumentException("trying to construct too large duration with " + nanos + "ns")
+    else if (nanos < 0)
+      fromNanos((nanos - 0.5).toLong)
```

Comment:
(double+0.5).toLong is broken anyway on a few `Double`s which `round` handles correctly, as @melrief reported on GitHub. Using `round` should fix the original issue as well, though I didn't test the integrated version.

See https://github.com/scala/scala/pull/5913 and https://github.com/scala/bug/issues/10320